### DISCUSSION TITLE
Can i add html headers in toc of mkdocs?

### DISCUSSION THREAD

**Discussion by asl369** [mkdocs/mkdocs#3038] [Q&A]
Hi, Will i be able to add html headers in toc of mkdocs?
For example
If i have #Heading1 and <h>Heading2<h> with html headers, is there any way that i could get Heading1 and Heading2 in TOC of mkdocs file?

**Comment by squidfunk** [comment]
The table of contents extension only processes Markdown headers, e.g.:

``` markdown
# Heading
## Subheading
...
```

If you're using HTML syntax to define headlines, the table of contents extension will ignore them.

``` html
<h1>Heading</h1>
<h2>Subheading</h2>
...
```

Thus, the answer to your question is no, the table of contents extension will ignore HTML headers.

**Comment by mh166** [comment]
Interestingly, I was about to ask the same. Knowing that it is not taking HTML headings into consideration by default, is there a way to add these to the parser / TOC manually?

I'm completely new to mkdocs, so I don't know if it's possible _at all_. But if so: do you have a hint on how to go about it? Like, is there a hook or a plugin API that would allow me to manually add entries to the TOC and link them to jump marks in the doc? I don't mind writing the code for that myself, but unfortunately I don't know where to start.

**Reply by squidfunk** [reply]
Maybe you leverage the new [hooks API](https://www.mkdocs.org/user-guide/configuration/#hooks).

**Reply by mh166** [reply]
Thanks, @squidfunk. That was just the hint I needed. 😃 Here is a proof of concept, adding a link to the TOC of a specific page:

```python
from mkdocs.structure.toc import AnchorLink

def on_page_content(html, page, **kwargs):
    if page.title != 'Newsletter - Week 39/2022':
        return
    for link in page.toc:
        link.children.append(AnchorLink('It works =)', 'it-works', 2))
```

As there is only one link directly in `page.toc`, I seem to get away with that `for ... in ...` loop. Should there ever be more, I'm probably screwed. 😁 But from what I've seen, this entry resolves to a heading of level 1 (think: `<h1>`) and [#818](https://github.com/squidfunk/mkdocs-material/issues/818#issuecomment-404156956) tells me that there can only be one `<h1>` per page/article. So I should be in the clear here.

Tomorrow, I'll refine this a little bit more: Now that the manual addition of a TOC entry works, I will make use of the `html` that is passed into the hook and add my custom headings as and where needed...

If you have any suggestions on how to improve this code, please feel free to tell me, as I'm only starting to learn Python. 🤓 Anyway, thanks again for your help. 👍

**Reply by squidfunk** [reply]
You can discriminate pages by their [`src_uri`](https://github.com/mkdocs/mkdocs/blob/56b235a8ad43f2300d17f87e6fa4de7a3d764397/mkdocs/structure/files.py#L151), might be more stable than using the title. It's equal to the location in `docs_dir`:

``` python
if page.file.src_uri == "path/to/index.md":
  ...
```

**Reply by mh166** [reply]
It appears to be `page.file.src_uri` instead, but that is indeed very helpful. Thanks for pointing that out! 😊

**Reply by squidfunk** [reply]
You're right, I've corrected the example. Good catch!
